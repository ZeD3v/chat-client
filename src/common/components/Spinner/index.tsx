import React from 'react';

import styles from './styles/Spinner.module.scss';

type SpinnerProps = {
  fullScreen?: boolean;
  overlay?: boolean;
  overlayColor?: string;
  hidden?: boolean;
  testId?: string;
};

const Spinner: React.FC<SpinnerProps> = ({
  fullScreen = false,
  overlay = false,
  overlayColor = '#fff',
  hidden = false,
  testId = undefined,
}: SpinnerProps) => {
  return (
    <div
      data-testid={testId}
      style={{
        background: overlayColor,
      }}
      className={`${styles.container}${
        fullScreen ? ` ${styles['container--fullscreen']}` : ''
      }${overlay ? ` ${styles['container--overlay']}` : ''}${
        hidden ? ` ${styles['container--hidden']}` : ''
      }`}
    >
      <div className={styles['inner-container']}>
        <div className={styles.sphere} />
        <div className={`${styles.sphere} ${styles['sphere--2']}`} />
      </div>
    </div>
  );
};

export default Spinner;
