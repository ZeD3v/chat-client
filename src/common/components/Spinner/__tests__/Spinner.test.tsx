import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Spinner from '../index';

describe('Modal', () => {
  test('renders correctly', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
