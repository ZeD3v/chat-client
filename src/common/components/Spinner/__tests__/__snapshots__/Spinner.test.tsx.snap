// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div>
  <div
    class="container"
    style="background: rgb(255, 255, 255);"
  >
    <div
      class="inner-container"
    >
      <div
        class="sphere"
      />
      <div
        class="sphere sphere--2"
      />
    </div>
  </div>
</div>
`;
